.github/workflows/qat.yml

```diff
@@ -14,12 +26,59 @@ permissions:
 jobs:
   playwright:
     name: Playwright
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     timeout-minutes: 90
-    if: false
+    env:
+      FEATURE_BRANCH_NAME: $(echo "${{ github.head_ref || github.ref_name }}" | sed -e 's/feature\/\(.*\)/\1/' | tr '[:upper:]' '[:lower:]')
+      NPM_QAT_SCRIPT: "qat:debug"
+    if: ${{ github.repository != 'penske-media-corp/github-workflows-wordpress' }}
 
     steps:
+      - name: Prepare environment
+        uses: penske-media-corp/github-action-wordpress-test-setup@main
+        with:
+          bitbucket_read_only_ssh_key: "${{ secrets.BITBUCKET_READ_ONLY_SSH_KEY }}"
+          github_read_only_ssh_key: "${{ secrets.GITHUB_READ_ONLY_SSH_KEY }}"
+          nodejs: 1
+
+      - name: Check QAT Script and Feature Branch
+        run: |
+          has_qat=`jq -r '.scripts|to_entries[]|select(.key=="${{ env.NPM_QAT_SCRIPT }}")|.key' package.json 2>/dev/null | head -n 1 | wc -l`
+          if [ $has_qat -eq 1 ] && [[ "${{ github.head_ref || github.ref_name }}" =~ 'feature/' ]]; then
+            echo "RUN_QAT=true" >> $GITHUB_ENV
+          else
+            echo "Not a feature branch or qat script not found. Skipping Playwright tests."
+            echo "RUN_QAT=false" >> $GITHUB_ENV
+          fi
+
+      - name: Check PMCQA
+        if: ${{ env.RUN_QAT == 'true' }} && ${{ inputs.THEME_NAME  != '' }} && ${{ inputs.SITE_URL != '' }}
+        timeout-minutes: 10
+        run: |
+          hash_url="https://${{ env.FEATURE_BRANCH_NAME }}.${{ inputs.SITE_URL }}/wp-content-vipgo-sites/${{ env.FEATURE_BRANCH_NAME }}/themes/vip/${{ inputs.THEME_NAME }}/version-hash"
+          echo "HASH_URL: $hash_url"
+          hash=`curl --insecure $hash_url`
+          until [[ $hash == ${{ github.sha }} ]]; do
+              echo -n "Waiting for Jenkins Deployment ... \n"
+              hash=`curl --insecure $hash_url`
+              sleep 5
+          done
+
+      - name: Install Playwright
+        if: ${{ env.RUN_QAT == 'true' }}
+        run: |
+          . "$NVM_DIR/nvm.sh"
+
+          nvm install
+          nvm use
+
+          npm ci
+          npx playwright install --with-deps
```

Comment:
Can we remove this line? since package.json loads playwright as a dep and we have `npm run qat` script we shouldn't need `npx`... right? https://github.com/penske-media-corp/pmc-sourcingjournal-2021/blob/main/package.json#L18

Reply:
I think the initial bitbucket setup copied what we do in checkout
* install playwright in CI https://github.com/penske-media-corp/subscription-checkout/blob/main/.github/workflows/main.yml#L54-L55 
* ...because playwright it not a dep https://github.com/penske-media-corp/subscription-checkout/blob/main/package.json

That decision was made because we don't have an `npm run qat` script in checkout. But now that we do have `npm run qat` in SJ we shouldn't need `npx` anymore. When we had playwright as a dep in checkout we had to use `npx playwright ..`, so we decided to remove it as a dep and install/use it globally, when really we should of just created `npm run qat`

Reply:
Thanks and you are right. I removed npx from here. (and I missed playwright is in package.json in SJ)